src/pobox.erl

```diff
@@ -198,21 +310,38 @@ notify(cast, _Msg, _State) ->
     keep_state_and_data;
 notify({call, From}, Msg, Data) ->    
     handle_call(From, Msg, notify, Data);
-notify(info, Msg, Data) ->    
+notify(info, Msg, Data) ->
     handle_info(Msg, notify, Data).
         
 
 %% @private
-handle_call(From, {resize, NewSize}, _StateName, S=#state{buf=Buf}) ->
-    {keep_state, S#state{buf=resize_buf(NewSize,Buf)}, 
-        [{reply, From, ok}]};
 handle_call(From, {post, Msg}, StateName, S=#state{buf=#buf{max=Size, size=Size}}) ->
     gen_server:reply(From, full),
     ?MODULE:StateName(cast, {post, Msg}, S);
 handle_call(From, {post, Msg}, StateName, S) ->
     gen_server:reply(From, ok),
     ?MODULE:StateName(cast, {post, Msg}, S);
-
+handle_call(From, {resize, NewSize}, _StateName, S=#state{buf=Buf}) ->
+    {keep_state, S#state{buf=resize_buf(NewSize,Buf)}, [{reply, From, ok}]};
+handle_call(From, {give_away, Dest, DestData, Origin}, _StateName, S=#state{owner=Owner, owner_pid=OwnerPid, name=Name}) ->
+    CanUsePid = case where(Dest) of
+        DestPid when is_pid(DestPid) ->
+            is_process_alive(DestPid) and (Origin =:= OwnerPid) and (DestPid =/= OwnerPid);
```

Comment:
If the process is alive and the origin of the call is the owner, and we aren't trying to transfer to ourselves, then transfer ownership and reply true, otherwise don't and reply false.